pkg/collection/entry/metadata_test.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package entry_test
+
+import (
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/collection/entry"
+)
+
+// TestMetadataSerialize verifies integrity of serialization.
+func TestMetadataSerialize(t *testing.T) {
```

Comment:
This test needs to be extended for the known edge cases with the encoding which are 0 character in the filename and in the mime type. I would rather avoid custom encodings.